llvm/lib/Target/AArch64/AArch64LoopIdiomTransform.cpp

```diff
@@ -187,6 +180,38 @@ AArch64LoopIdiomTransformPass::run(Loop &L, LoopAnalysisManager &AM,
 //
 //===----------------------------------------------------------------------===//
 
+struct MemCompareIdiom {
+private:
```

Comment:
Could we use class here instead?

Reply:
We could, but I saw lots of other places in LLVM where we use struct for smaller tasks like this so I thought maybe I should do the same. To be honest, I'm not entirely sure what the difference is. I'm happy to change it to class if that brings benefits!